### PR TITLE
Hw 6.class

### PR DESCRIPTION
**Pull request by ninaprus**
class Ip
class FileTools
class Unit

### CODE REVIEW

File: hw_7.class.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# 2) Чтение из файла
+# 3) Объединение данных из файлов в новый файл
+# 4) Получить путь относительный путь к файлу
+# 5) Получить абсолютный путь к файлу
+#
+def get_file_tools(my_file):
+    """ Function creates an instance of the FileTools class.
+    Actions: reading and writing to a file, merging files
+    getting relative and absolute paths"""
+    import os.path
```
Comment (katevoskresenska) [inline hw_7.class.py]: It is a more pythonic way to declare imports at the beginning of the file

File: hw_7.class.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    if os.path.exists(my_file):
+
+        class FileTools:
```
Comment (katevoskresenska) [inline hw_7.class.py]: I think it is a not very good idea to create a class inside the function as  the class will be re-costructed at every function call. You can check if the file exists inside the class using try\except construction.

File: hw_7.class.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+            def write_this(self, text):
+                with open(self._file_1, 'w') as writer:
+                    writer.write(text)
+
+            def read_it(self):
+                with open(self._file_1, 'r') as reader:
+                    for line in reader:
+                        print(line, end='')
+
```
Comment (katevoskresenska) [inline hw_7.class.py]: It would be nice to add one more method for JSON validation to catch JSONDecodeError

File: hw_7.class.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+# 4) Получить список последних октетов IP адресов
+# (10.11.12.13 -> 13)
+class Ip:
+    def __init__(self, ip):
+        self._ip = ip
+
+    def get_ip(self):
+        return self._ip
+
+    def get_ip_expand(self):
```
Comment (mmykhailenko) [inline hw_7.class.py:18]: your methods should return something

File: hw_7.class.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+            def __init__(self, file_1):
+                self._file_1 = file_1
+
+            def write_this(self, text):
+                with open(self._file_1, 'w') as writer:
+                    writer.write(text)
+
+            def read_it(self):
+                with open(self._file_1, 'r') as reader:
```
Comment (mmykhailenko) [inline hw_7.class.py]: check if file exists, before open for reading

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**